File: api/src/main/kotlin/com/api/exceptionhandler/ExceptionResponse.java
package com.api.exceptionhandler;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class ExceptionResponse {

    private String code;
    private String message;

    public ExceptionResponse(CommonExceptionType type) {
        this.code = type.getCode();
        this.message = type.getMessage();
    }

    public ExceptionResponse(CommonExceptionType type, Exception e) {
        this.code = type.getCode();
        this.message = type.getMessage() + ", " + e.getMessage();
    }

    public ExceptionResponse(CommonException e) {
        this.code = e.getType().getCode();
        this.message = e.getMessage();
    }

    public ExceptionResponse(CommonExceptionType type, String message) {
        this.code = type.getCode();
        this.message = type.getMessage()+ ", " + message;
    }
}

File: api/src/main/kotlin/com/api/exceptionhandler/ExceptionResponse.kt
package com.api.exceptionhandler

import com.core.component.exception.CommonException
import com.core.component.exception.CommonExceptionType
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@Getter
@Setter
@NoArgsConstructor
class ExceptionResponse {
    private var code: String?
    private var message: String?

    constructor(type: CommonExceptionType) {
        this.code = type.code
        this.message = type.message
    }

    constructor(type: CommonExceptionType, e: Exception) {
        this.code = type.code
        this.message = type.message + ", " + e.message
    }

    constructor(e: CommonException) {
        this.code = e.type?.code
        this.message = e.message
    }

    constructor(type: CommonExceptionType, message: String) {
        this.code = type.code
        this.message = type.message + ", " + message
    }
}
